ggshield/core/git_shell.py

```diff
@@ -125,15 +130,38 @@ def git_ls(wd: Optional[str] = None) -> List[str]:
     return shell_split(cmd, timeout=600)
 
 
+def is_valid_git_commit_ref(ref: str) -> bool:
+    """
+    Check if a reference is valid and can be resolved to a commit
+    """
+    ref += "^{commit}"
+    cmd = [GIT_PATH, "cat-file", "-e", ref]
+
+    try:
+        shell(cmd, raise_on_process_error=True)
+    except subprocess.CalledProcessError:
+        return False
+
+    return True
+
+
 def get_list_commit_SHA(commit_range: str) -> List[str]:
     """
     Retrieve the list of commit SHA from a range.
     :param commit_range: A range of commits (ORIGIN...HEAD)
     """
+    try:
+        commit_list = shell_split(
+            [GIT_PATH, "rev-list", "--reverse", *commit_range.split(), "--"],
+            raise_on_process_error=True,
+        )
+    except subprocess.CalledProcessError as e:
+        if b"bad revision" in e.stderr and commit_range.endswith("~1..."):
+            commit_ref = commit_range[:-5]
```

Comment:
Can you add a comment explaining we handle the case where `commit_ref` has no parent here?